app/controllers/admin/tour_details_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Admin::TourDetailsController < ApplicationController
+  layout "admin/layouts/admin_layout"
+  before_action :check_admin, only: %i(show edit update destroy)
```

Comment:
```suggestion
  before_action :check_admin, except: %i(new create)
```

Reply:
Mà check quyền admin thì phải check hết toàn bộ các action chứ sao lại check mỗi mấy action kia nhỉ 🤔